service/src/main/kotlin/fi/espoo/evaka/reports/HolidayPeriodAttendanceReport.kt

```diff
@@ -0,0 +1,342 @@
+// SPDX-FileCopyrightText: 2017-2024 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.reports
+
+import fi.espoo.evaka.Audit
+import fi.espoo.evaka.absence.getAbsences
+import fi.espoo.evaka.assistance.getAssistanceFactorsForChildrenOverRange
+import fi.espoo.evaka.attendance.occupancyCoefficientSeven
+import fi.espoo.evaka.backupcare.getBackupCaresForDaycare
+import fi.espoo.evaka.daycare.getDaycare
+import fi.espoo.evaka.document.childdocument.ChildBasics
+import fi.espoo.evaka.holidayperiod.getHolidayPeriod
+import fi.espoo.evaka.placement.PlacementType
+import fi.espoo.evaka.reservations.getReservationBackupPlacements
+import fi.espoo.evaka.reservations.getReservations
+import fi.espoo.evaka.serviceneed.ShiftCareType
+import fi.espoo.evaka.shared.DaycareId
+import fi.espoo.evaka.shared.HolidayPeriodId
+import fi.espoo.evaka.shared.PersonId
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.db.Predicate
+import fi.espoo.evaka.shared.domain.BadRequest
+import fi.espoo.evaka.shared.domain.EvakaClock
+import fi.espoo.evaka.shared.domain.FiniteDateRange
+import fi.espoo.evaka.shared.domain.getHolidays
+import fi.espoo.evaka.shared.domain.getOperationalDatesForChildren
+import fi.espoo.evaka.shared.security.AccessControl
+import fi.espoo.evaka.shared.security.Action
+import java.time.LocalDate
+import java.time.Period
+import kotlin.math.ceil
+import org.jdbi.v3.core.mapper.Nested
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class HolidayPeriodAttendanceReport(private val accessControl: AccessControl) {
+    @GetMapping("/employee/reports/holiday-period-attendance")
+    fun getHolidayPeriodAttendanceReport(
```

Comment:
Whew, that's a lot of code. :D Made it only half way through before a headache got worse. If you'd like a proper review for this function then let's maybe do it in a small group over a call, but I guess this should be quite safe to test in production too. Not sure if something could be done to make it more digestible. Probably the business logic simply is very complex.